### DISCUSSION TITLE
Copilot loses functionality

### DISCUSSION THREAD

**Discussion by jasonvagner** [logancyang/obsidian-copilot#95] [Q&A]
Howdy.

Quite liking this plugin, but I had some stability issues today. I installed it, entered my OpenAI API key, and ran some basic commands: convert selection to tweet thread, shorten/lengthen, and then I fed some prompts in the conversation sidebar.

I was so pleased with the results, I went to show someone else how well it worked.

And then Copilot wouldn't behave whatsoever.
- After selecting text and typing control-P to bring up the command list, there were only a few Copilot commands in the list, none of the ones I wanted. I tried shortening, lengthening and changing the selection, to no avail.
- I restarted the plugin, reverted to defaults, re-entered API key, turned things off and on again, and this time it had the "convert to tweet" command (but not most of the others) but the returned results were in kanji.
- I restarted, reverted defaults, re-entered API key, etc., restarted Obsidian, and now things seem to be working again.

I turned on debug mode but don't know where the output is.

Is this variability common at this stage of development?
Any more direct recommendation on resolving the issue when it arises?

Thanks,
Jason

**Comment by logancyang** [comment]
Hmm, I only encountered this when I was actively updating the code during development without restarting the plugin. This seems to be an Obsidian issue and not the plugin itself. I've never seen this outside of development. I don't think it's necessary to reset everything, just restart Obsidian.

I wonder what else you did to get this behavior? If I can reproduce it I can understand what's happening better.

Don't think this helps cuz it's not likely a plugin issue but since you asked - the debug messages are in the console. I doubt it would print anything useful for this particular issue though.

**Comment by logancyang** [comment]
Oh I forgot to reply here. It was because you don't have the chat window focused. Once you have it in focus the commands show up correctly.
